src/clr-angular/utils/popover/popover-content.ts

```diff
@@ -52,6 +53,11 @@ export class ClrPopoverContent implements AfterContentChecked, OnDestroy {
     this.smartEventsService.scrollToClose = !!scrollToClose;
   }
 
+  @Input('clrPopoverContentSmartPositioning')
```

Comment:
I don't see the need for this `setter`;

```ts
@Input('clrPopoverContentSmartPositioning') smartPositioning: boolean;
```
It will do the same job. This also will remove the need for `this. _smartPositioning`. 
I'm left with the idea that we are going away from adding `_` as the prefix to indicate private variables and use them only in the cases when we are creating `setter/getter` with the same name.
```ts
private _open: boolean;
set open(v: boolean) { /* some code transformation */ }
get open(): boolean { /* some code transformation */ }
```

Reply:
I also had this thought at some point, but intentionally decided against it.
With this syntax it remains fixed only as input parameter. If I expose it as something that the user may read through the typescript API, then for completeness I'd better also expose smartPositioningChange observable, or maybe even an Output. For now I find this to be obsolete, as this variable can't change internally in the component and I prefer to keep the possible interactions only to the intended cases, with one-way input binding.
Hope this makes sense?